src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/viewmodel/RecordableViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel
+
+import com.github.thomasnield.rxkotlinfx.observeOnFx
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleObjectProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.workbook.AssociatedAudio
+import org.wycliffeassociates.otter.common.data.workbook.Take
+import org.wycliffeassociates.otter.common.domain.content.EditTake
+import org.wycliffeassociates.otter.common.domain.content.RecordTake
+import org.wycliffeassociates.otter.common.domain.content.Recordable
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.TakeContext
+import tornadofx.FX.Companion.messages
+import tornadofx.*
+
+open class RecordableViewModel(private val audioPluginViewModel: AudioPluginViewModel) {
+
+    val recordableProperty = SimpleObjectProperty<Recordable?>()
+    var recordable by recordableProperty
+
+    private val disposables = CompositeDisposable()
+
+    val selectedTakeProperty = SimpleObjectProperty<Take?>()
+    val contextProperty = SimpleObjectProperty<TakeContext>(TakeContext.RECORD)
+    val showPluginActiveProperty = SimpleBooleanProperty(false)
+    var showPluginActive by showPluginActiveProperty
+
+    val snackBarObservable: PublishSubject<String> = PublishSubject.create()
+
+    val alternateTakes: ObservableList<Take> = FXCollections.observableList(mutableListOf())
+
+    init {
+        recordableProperty.onChange {
+            // TODO
+        }
+
+        selectedTakeProperty.onChange {
+            // TODO
+        }
+    }
+
+    fun recordNewTake() {
+        recordable?.let {
+            contextProperty.set(TakeContext.RECORD)
+            showPluginActive = true
+            audioPluginViewModel
+                .record(it)
+                .observeOnFx()
+                .subscribe { result ->
+                    showPluginActive = false
+                    when (result) {
+                        RecordTake.Result.NO_RECORDER -> snackBarObservable.onNext(messages["noRecorder"])
+                        RecordTake.Result.SUCCESS, RecordTake.Result.NO_AUDIO -> {}
+                        null -> {} // This cannot happen but the compiler complains if null branch does not exist
```

Comment:
As in another review, this is avoidable if the type is specified as non-null for JVM interop. 

Same as there, I don't think this is worth another commit, but fix it if you wish.

Reply:
Perhaps it's not worth another commit, but I was too curious to not try it. I can't get it to work though- do you happen to know how? (I can't apply the `@NotNull` annotation to an enum class)

Reply:
Never mind- I figured out that you can specify the type when you name the parameter in the callback. Thanks @aunger!